File: script/solutions/fibonacci_level2_sol.py
# Ploblem : [피보나치 수 4] https://www.acmicpc.net/problem/10826

# Solver : Seonghwan Lee
# Solved Date : 2021-02-24
# BigO: N


def fibonacci_2(n):
    if n == 0:
        return 0
    elif n == 1:
        return 1

    ret_result = [1,1]
    for i in range(2,n):
        ret_result.append(ret_result[i-1] + ret_result[i-2])
    return ret_result[-1]


if __name__ == '__main__':
    n = int(input())
    print(fibonacci_2(n))

File: script/solutions/fibonacci_level2-sol.cpp
// Ploblem: [피보나치 수 4] https://www.acmicpc.net/problem/10826

// Solver: Seonghwan Lee
// Solved Date: 2021-03-08
// BigO: N

#include<iostream>
using namespace std;

unsigned long long fib(int n){
    unsigned long long arr[n+1];
    arr[0] = 0;
    arr[1] = 1;

    for (int i = 2; i < n+1; i++){
        arr[i] = arr[i-1] + arr[i-2];
    }
    return arr[n];
}

int main(){
    int n;
    scanf("%d", &n);
    printf("%llu", fib(n));
    return 0;
}
